### DISCUSSION TITLE
How to get Android activity/application context?

### DISCUSSION THREAD

**Discussion by hepspl** [skiptools/skip#122] [Q&A] [answered]
I am writing a multiplatform secure storage manager class which will use Keychain on iOS and Encrypted Shared Preferences on Android.
`EncryptedSharedPreferences.create()` requires activity or application context handle.

I wonder how can I get such handle in typical non-Context class which I am writing in Swift...
Unfortunately I am not an Android developer so maybe I am missing some easy static way to get the context.

I was thinking about using `getApplicationContext()` but I think it should be exposed somewhere in Skip generated Android code to be used in any class created later in common source folder... I don't really know how to do it.

Any help appreciated.

**Comment by marcprux** [comment, marked as answer]
For Skip-generated projects, we stash the global application context in the `Foundation.ProcessInfo` singleton, which you can access like this:

```swift
import Foundation

#if SKIP 
let appContext = ProcessInfo.processInfo.androidContext
#endif
```
